lib/adapter.js

```diff
@@ -624,10 +626,21 @@ function Adapter(options) {
             throw new Error('checkPassword: no callback');
         }
 
-        user = (user || '').toString().replace(this.FORBIDDEN_CHARS, '_').replace(/\s/g, '_').replace(/\./g, '_').toLowerCase();
```

Comment:
Do we now allow forbidden chars, spaces and dots in usernames?

Reply:
Have they ever been forbidden in usernames? I don't think so. At least you can easily create users with dots etc. in the username via admin

Reply:
I'm not entirely sure, doesn't the username translate into a user object id? And the characters in FORBIDDEN_CHARS and spaces have been forbidden in object IDs for a while now. If its possible to create users with those names, we should raise an issue in admin IMO.

Reply:
@Apollon77 ?

Reply:
this is what this PR is about, we no longer want to guess the id by the username how it is done atm, because it's also possible to have ids that have nothing to do with the username, , usernames with different forbidden chars can leadd to the same id, admin allows to configure id manually etc.. Thus, we want to cache the id related to every username.

Reply:
Then we need to make sure that the ID is valid IMO

Reply:
due to the issues with github the commit is not displayed here yet, but theres an adaption in my fix-user branch, so probably just a matter of time

Reply:
And what about whitespace chars? `\s`

Reply:
Are whitespaces forbidden in ids?

Reply:
AFAIK yes. @Apollon77 ???

Reply:
ok nevermind: https://github.com/ioBroker/ioBroker.js-controller/issues/341#issuecomment-527851885

Reply:
LGTM then.

Reply:
sorry for answering nmow ... everything correct :-)